firecracker-control/local.go

```diff
@@ -224,6 +255,7 @@ func (s *local) startMachine(ctx context.Context, id string, machine machineIfac
 // StopVM stops running VM instance by VM ID
 func (s *local) StopVM(ctx context.Context, req *proto.StopVMRequest) (*empty.Empty, error) {
 	logger := log.G(ctx).WithField("vm_id", req.GetVMID())
+	s.cleanup(logger)
```

Comment:
This will do the clean up for all VMs, right? We don't want that.

Reply:
Naw, this will clean up everything associated with this particular VM. In this case, the stub drive files associated with the VM.

Reply:
Can you clarify how it works? I see that in `buildVMConfiguration` you do:
```
s.cleanupFns = append(s.cleanupFns, func() error {return os.RemoveAll(tempDir)})
```
and inside `cleanup` you iterate over the whole list and invoke cleanup func.
```
for _, fn := range s.cleanupFns { fn(); }
```

so if we have:
createVM(1)
createVM(2)
stopVM(1)
you will do a cleanup for both VMs.

question 2: `cleanupFns` not thread safe.

Reply:
@mxpv - Ah, maybe this is a misunderstanding on my end. I thought every time `CreateVM` was called a new local would be created, is that not true?

Reply:
No. containerd plugins are singletons.